Topics/Tech_Stacks/PostgreSQL_psycopg2.md

```diff
@@ -45,81 +45,95 @@ A quick and basic runover of PostgreSQL.
-	password VARCHAR ( 50 ) NOT NULL,
-	email VARCHAR ( 255 ) UNIQUE NOT NULL
+CREATE TABLE teams (
+	team_id SERIAL PRIMARY KEY,
+	team_name VARCHAR(50) NOT NULL,
+	sport_id INT NOT NULL,
+	FOREIGN KEY (sport_id) REFERENCES sports(sport_id),
+);
+```
+```
+CREATE TABLE players {
+	player_id SERIAL PRIMARY KEY,
+	first_name VARCHAR(50) NOT NULL,
+	last_name VARCHAR(50) NOT NULL,
+	age VARCHAR(50) NOT NULL,
+	team_id INT NOT NULL,
+	FOREIGN KEY (team_id) REFERENCES teams(team_id),
 );
 ```
-The text after the datatypes (e.g. UNIQUE NOT NULL) are optional.
+The text after the datatypes (e.g. `UNIQUE NOT NULL`) are optional [constraints](https://www.postgresql.org/docs/16/ddl-constraints.html#DDL-CONSTRAINTS) that define conditions that must be satisfied by the data.
 
-Here is a link for information on datatypes in PostgreSQL: https://www.postgresql.org/docs/current/datatype.html
+Here is a link for information on datatypes in PostgreSQL: [https://www.postgresql.org/docs/current/datatype.html](https://www.postgresql.org/docs/current/datatype.html)
 
-Here is a link for more details and examples of how to create a table: https://www.postgresql.org/docs/current/sql-createtable.html
+Here is a link for more details and examples of how to create a table: [https://www.postgresql.org/docs/current/sql-createtable.html](https://www.postgresql.org/docs/current/sql-createtable.html)
 
 
 ### Table Operations
-Moving onto table operations. To select data from a certain column in a table, you can use a query like:
+In SQL, a query is a statement/command used to retrieve or manipulate data in a relational database. To retrieve data from a certain column in a table, you can use the `SELECT` statement and `FROM` clause:
 ``` 
-    SELECT col_name1, col_name2 FROM table_name;
+    SELECT first_name, last_name FROM players;
 ```
-This gives you a smaller table that only contains the columns "col_name1" and "col_name2" from your selected table in your relation. 
+The output of this query is a smaller table that only contains the columns `first_name` and `last_name` of each row from the table `players`.
 
-You can also merge two different tables by changing the FROM clause to contain 2 tables:
+You can also take the cross product of two or more tables by listing the tables in the `FROM` clause:
 ```
-    SELECT col1_table1, col2_table2 FROM table1_name, table2_name;
+    SELECT sport_name, team_name FROM sports, teams;
 ```
-This operation, first, performs a cross product between table1 and table2, where every row of one table is matched with every other row of the other table. Then it selects the data present in the new cross-products table from columns col1_table1 and col2_table2.
+First, this query performs a cross product between `sports` and `teams`. Each row of table `sports` is matched with each row of the table `teams`. Then, it selects the data from the columns `sport_name` and `team_name` of every row present in the newly joined table.
```

Comment:
What exactly does match mean? Which attributes are included in the result? I think an example would provide more value to explain this concept over words.

Reply:
I craeted new instances for each table and also added more details for the explanations. Can you check if it looks better now?